app/src/main/java/nu/yona/app/api/manager/impl/ActivityCategoryManagerImpl.java

```diff
@@ -120,4 +120,29 @@ public void onError(Object errorMessage) {
         }
     }
 
+    public void updateNetworkAPIEnvironment(){
+        activityCategoriesNetwork.updateNeworkEnvironment();;
+    }
+
+    public void validateNewEnvironment(String url,final DataLoadListener listener) {
+        try {
+            activityCategoriesNetwork.getActivityCategories(url, new DataLoadListener() {
```

Comment:
Why are you wrapping the listener you got?

Reply:
It is a mistake. I will revisit the code part.

Reply:
it is for sending back the response to the View class.

Reply:
But it doesn't do anything except for the null-checks, right? If you put the null-check inside ``getActivityCategories``, you don't need to do it here.

Reply:
It passes on the result with listener.onDataLoad(result) and listener.onError(errorMessage). 
or did I not get your question properly>

Reply:
What would go wrong if you would only do this?
```
try {
	activityCategoriesNetwork.getActivityCategories(url, listener);
} catch (Exception e) {
	AppUtils.throwException(ActivityCategoryManagerImpl.class.getSimpleName(), e, Thread.currentThread(), listener);
}
```